### ISSUE TITLE
Accessibility markup problem in Carbon overflow menu example

### ISSUE THREAD

**Issue by carmacleod** [carbon-design-system/carbon#2273]
In the [example Carbon vanilla overflow menu](https://www.carbondesignsystem.com/components/overflow-menu/code):
- the `div` with `role="button"` and the `ul` with `role="menu"` need to be siblings
(to remove a violation in DAP because buttons aren't supposed to have list children).

**Comment by carmacleod** [comment]
Note that the DAP violation occurs before the overflow menu is dropped down for the first time, i.e. the initial DOM needs to be changed so that the `ul` is not a child of the `div` with `role="button"`.

However, please ensure that even after the menu is dropped down, the `ul` should immediately follow the div with role button in the DOM, so that screen reader users can use their virtual cursor to read through the menu items without getting totally lost.

**Comment by emyarod** [comment]
related https://github.com/carbon-design-system/carbon/issues/2495

**Comment by elizabethsjudd** [comment]
@emyarod if you want to assign this to me, I can work on it next week.

**Comment by emyarod** [comment]
@elizabethsjudd looks like I'm not able to make that assignment

![image](https://user-images.githubusercontent.com/8265238/59945063-097a6980-942c-11e9-9917-f072cdb451e2.png)

**Comment by elizabethsjudd** [comment]
@emyarod, @mattrosno is reaching out to get this updated. Thanks!

**Comment by elizabethsjudd** [comment]
@carbon-design-system/developers used `related to` instead of `resolves` in PR #3181  but this issue is complete now
